lib/OpenQA/WebAPI/Controller/API/V1/Search.pm

```diff
@@ -117,6 +117,23 @@ sub query {
         }
         $cap -= scalar @results;
         last if $cap < 1;
+
+        # Job templates
+        my $last_group = undef;
+        my $like       = {like => "%${keywords}%"};
+        my $templates  = $self->schema->resultset('JobTemplates')
+          ->search({-or => {name => $like, description => $like}}, {limit => $cap});
+        while (my $template = $templates->next) {
+            my $contents = $template->name . "\n" . $template->description;
+            if ($template->group->id == $last_group) {
```

Comment:
@kalikiana  I did not realize this $last_group would be `undef` until I did a test.
There is a message when I use the search function: `Use of uninitialized value $last_group in numeric eq (==) at /home/liuxiaojing/code/repos/openQA/script/../lib/OpenQA/WebAPI/Controller/API/V1/Search.pm line 128.`

Reply:
@Amrysliu is right. If one only looks at the diff it isn't very apparent that this is actually within the loop. This function really gets a little bit big.

Reply:
I proposed #3335 as a follow-up, refactoring the result types into separate functions and addressing the above concerns in tandem.